sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs

```diff
@@ -46,9 +56,49 @@ pub fn tendermint_state_proof(
     ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    todo!()
+    let value = pop_arg!(args, Vector).to_vec_u8()?;
+    let path = pop_arg!(args, Vector).to_vec_u8()?;
+    let prefix = pop_arg!(args, Vector).to_vec_u8()?;
+    let root = pop_arg!(args, Vector).to_vec_u8()?;
+    let proof = pop_arg!(args, Vector).to_vec_u8()?;
+
+    let Ok(prefix) = CommitmentPrefix::try_from(prefix) else {
+        return Ok(NativeResult::err(context.gas_used(), 1));
```

Comment:
please don't use magic number (1) -- define constants. Make sure we have all error codes unique in the whole package.

Reply:
so probably we should have `light_client/errors.rs` module.

Reply:
Yes! I add todo for this. Will do it after merge,.